plugins/chat/app/services/update_channel.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+module Chat
+  module Service
+    # Service responsible for updating a chat channel's name and description.
+    #
+    # For a CategoryChannel, the settings for auto_join_users and allow_channel_wide_mentions
+    # are also editable.
+    #
+    # @example
+    #  Chat::Service::UpdateChannel.call(channel: channel, guardian: guardian)
+    #
+    class UpdateChannel
+      include Base
+
+      # @!method call(channel:, guardian:)
+      #   @param [ChatChannel] channel
+      #   @param [Guardian] guardian
+      #   @return [Context]
+
+      before_contract { guardian(:can_edit_chat_channel?) }
+
+      before_contract do
+        context.name = (context.name || context.channel.name).presence
```

Comment:
This is nicer, though I think still not quite right, since there is no way to even blank out name or description if necessary (though it's much less likely to want to do this for name). This is where the contortions I introduced helped, since you could inspect the original params passed in and see if they were even present. If description was present + nil, then we could blank it out, otherwise if it was not present we would not touch it.

Not sure how to express this without having the original params...

Reply:
Yes this is all messed up ATM, feel free to fix if you want, I noted to come back to this tomorrow.

Reply:
@jjaffeux I made this commit https://github.com/discourse/discourse/pull/19814/commits/ecf57a1062e68884d53b8136dcc3216dc11da595 to address this, I made the `@initial_context` hash available to the service so it can be inspected in `before_contract`.